quic/s2n-quic-tls/src/tests.rs

```diff
@@ -249,6 +268,28 @@ fn s2n_server_with_client_auth() -> Result<server::Server, Error> {
         .build()
 }
 
+fn s2n_server_with_client_auth_optional() -> Result<server::Server, Error> {
+    server::Builder::default()
+        .with_client_auth_type(ClientAuthType::Optional)?
+        .with_empty_trust_store()?
+        .with_client_authentication()?
+        .with_verify_host_name_callback(VerifyHostNameClientCertVerifier::new("qlaws.qlaws"))?
+        .with_certificate(CERT_PEM, KEY_PEM)?
+        .with_trusted_certificate(CERT_PEM)?
+        .build()
+}
+
+fn s2n_server_with_client_auth_none() -> Result<server::Server, Error> {
+    server::Builder::default()
+        .with_empty_trust_store()?
+        .with_client_authentication()?
+        .with_verify_host_name_callback(VerifyHostNameClientCertVerifier::new("qlaws.qlaws"))?
+        .with_certificate(CERT_PEM, KEY_PEM)?
+        .with_trusted_certificate(CERT_PEM)?
+        .with_client_auth_type(ClientAuthType::None)?
```

Comment:
Did you mean to call `with_client_auth_type` after `with_certificate`. In fact I think there are probably two cases to test, calling before and calling after to verify that behavior is the same. I also wonder if there is a better way to test this and avoid alot of this boilerplate duplication.